### PR TITLE
Tools/ApplicationDebugger: update for the 2023.0

### PR DESCRIPTION
**Pull request by jimmytwei**
Signed-off-by: Saiapova, Natalia <natalia.saiapova@intel.com>

# Existing Sample Changes
## Description

Update for the compiler 2023.0;
Update for the debugger 2023.0:
- on Windows we do not support debugging the code offloaded on GPU
- update supported platforms and OS
- for Jacobi sample, it makes sense to focus only on Linux* OS at the moment, so removed all Windows files.

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality): update to new compiler and debugger
- [ ] Implement fixes for ONSAM Jiras

## How Has This Been Tested?

I tested it locally on my Linux* 20.04 with jit and AoT modes.

- [x] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode
- [x] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

FYI @barisaktemur, @jimmytwei

### CODE REVIEW

**Comment by jimmytwei** [comment]
@barisaktemur Can you please review this PR? There were some conflicts in #1206 that prevented merging. It's the exact same changes. 

@praveenkk123 Can you approve and merge this once @barisaktemur approves?

**Review by praveenkk123 — APPROVED** [review]

**Pull request merged**
